### DISCUSSION TITLE
App-specific window positions / sizes for window throw, etc.

### DISCUSSION THREAD

**Discussion by galligan** [rxhanson/RectanglePro-Community#213] [Ideas]
Hi! I only just got started with Rectangle Pro in the last week and have been loving it so much. Previously I've used Keyboard Maestro and Moom for similar things, but Rectangle Pro has blown well past those for me. Window throw especially has been an incredible new addition to my productivity.

One thing that I encounter regularly is that for some apps—let's say Messages, Mail, Slack, etc.—there are many window positions / sizes that I would never want them in or have them in. Similarly, there may be spots that I'd never have my browser in.

What I'd love to see is that I could create a "folder" of apps that have different behavior applied whenever I try to do window throw or something with them. This would just further amplify what I can already do with Rectangle since I could effectively map a gesture to the most common spot I always put a particular app, and have that same shortcut or gesture be different for the other apps.

Appreciate all that you've done with this! Can't wait to see what else comes.

**Comment by rxhanson** [comment]
Thanks for the feedback! 

I've thought about window throw profiles per display, but haven't actually thought about profiles per app - interesting idea. It might be tricky to make configuring something like this simple and easy to communicate so I'll have to give this some thought.

In the meantime, it might interest you to try out the Snap Targets feature if you haven't already done so. The reason I suggest this is because if you make an App Layout and within each configuration you check the box for "Use as a snap target" then that snap target is only displayed for windows of the configured app. You can use the "move" functionality configured in the Window Throw tab to activate a snap target, making it faster than dragging a window by the title bar. It's not quite what you've suggested, but it might fit in with your workflow.

**Reply by galligan** [reply]
Yeah I'm not 100% sure of the best way to communicate this, though I did basically get something similar accomplished based on what you'd described with App Layouts. I hadn't considered that before!

I set up one App Layout configuration called "Home Positions" with a keyboard shortcut, and only selected "Only execute on frontmost app". Then I configured each app's designated home position and `use as snap target`, along with some sequential execution changes if applicable. Here's what that looked like:

<img width="888" alt="image" src="https://user-images.githubusercontent.com/25037981/228282581-9ee25744-0245-420f-a518-790ae0850ccb.png">

What's great about this is:

- I have a singular key to fire on any app and have it set up in the exact spot I want
- I set up a mouse trigger on my Logitech so that I can gesture and fire the same key

Now that I know this is possible, Rectangle Pro seems even more powerful than before! Great stuff, and thanks!

**Reply by rxhanson** [reply]
Awesome, thanks for sharing your setup! I'm going to make some tutorial videos in the near future and will include this feature combination.
